### PR TITLE
Sync palindrome products exercise with problem specification

### PR DESCRIPTION
**Pull request by themetar**
[As discussed on the [forum](https://forum.exercism.org/t/why-is-the-palindrome-products-exercise-not-in-sync-with-problem-spec/13149)]

* Added the missing test cases
* Updated the example solution to comply to new tests

### CODE REVIEW

File: exercises/practice/palindrome-products/.meta/example.rb [hunk 1]
```diff
@@ -1,43 +1,40 @@
...
 
 class Palindromes
-  attr_reader :range
-  def initialize(options)
-    max = options.fetch(:max_factor)
-    min = options.fetch(:min_factor) { 1 }
-    @range = (min..max)
-  end
+  private
+  attr_reader :min, :max
```
Comment (kotp) [inline exercises/practice/palindrome-products/.meta/example.rb:5]:
```suggestion
  private
  
  attr_reader :min, :max
```

File: exercises/practice/palindrome-products/.meta/example.rb [hunk 2]
```diff
@@ -1,43 +1,40 @@
...
+    enum.lazy.
+      filter { |number| palindrome? number }.
+      map { |number| Palindrome.new number, factors(number) }.
+      find { |palindrome| !palindrome.factors.empty? }
   end
 
-  def smallest
-    sort.first[1]
+  public
+  attr_reader :smallest, :largest
```
Comment (kotp) [inline exercises/practice/palindrome-products/.meta/example.rb:33]:
```suggestion
  public
  
  attr_reader :smallest, :largest
```

**Review by kotp — CHANGES_REQUESTED** [review]
You should be able to apply the suggestions, they are simply to provide a blank line after visibility methods `private` and `public` as they do not affect only the following line.

File: exercises/practice/palindrome-products/.meta/example.rb (hunk 1 above)
Comment (themetar) [reply exercises/practice/palindrome-products/.meta/example.rb:5]: vscode kept triggering rubocop and auto-formatting my code. I assumed that that's exercism's style.

I think this is the rule that's being followed:
https://github.com/exercism/ruby/blob/fa35ccc41372394db18b4c4a140e27b2597d38d6/.rubocop.yml#L38

**Comment by themetar** [comment]
I've added the blank lines. @kotp

**Review by kotp — APPROVED** [review]

**Pull request merged**